### DISCUSSION TITLE
Smithy cli behind an HTTP proxy

### DISCUSSION THREAD

**Discussion by Robert904** [awslabs/smithy#2024] [Q&A] [answered]
Hi !

I try to install smithy cli on a windows computer (please, don't judge me...), reaching internet through an authenticating http proxy.

I managed to have it working for the git part (smithy init), but I can't get through for dependencies collection... The apache http library does not seem to use the proxy parameters I pass to the JVM (using the SMITHY_OPTS environment variable):
`set SMITHY_OPTS=-Dhttp.proxyHost=%proxy_host% -Dhttp.proxyPort=%proxy_port% -Dhttp.proxyUser=%proxy_user% -Dhttp.proxyPassword=%proxy_passwd% -Dhttps.proxyHost=%proxy_host% -Dhttps.proxyPort=%proxy_port% -Dhttps.proxyUser=%proxy_user% -Dhttps.proxyPassword=%proxy_passwd%`

These parameters are indeed passed to the JVM (checked using DEBUG=true) but looking at the --debug output it still tries to go directly to internet and ultimately fails to collect dependencies.

Any other way to configure the proxy for Smithy to use ? I can't find anything in the docs or in the existing issues...

Many thanks

**Comment by hpmellema** [comment, marked as answer]
The following pr adds support for using a proxy for smithy dependency resolution: https://github.com/smithy-lang/smithy/pull/2076
This functionality will be included in the next release. 

As noted in the PR, use of a proxy would be supported in 2 ways: 
1. Via environment variables: SMITHY_PROXY_HOST and SMITHY_PROXY_CREDENTIALS . These will set a common proxy configuration for all repositories
2. Via a specific configuration on the Maven repository definition. For example,
```
    "maven": {
        "repositories": [
            {
                "url": "http://localhost:1234/maven/not/there",
                "proxyHost": "http://localhost:2323",
                "proxyCredentials": "user:pass"
            }
        ],
        "dependencies": [
            "software.amazon.smithy:smithy-aws-iam-traits:1.26.0"
        ]
    }
 ```
